### DISCUSSION TITLE
Installation of Homebrew gets stuck at [Receiving Objects] stage

### DISCUSSION THREAD

**Discussion by KernelBypass** [Homebrew/discussions#3680] [Getting started] [answered]
I am new to brew. I tried running:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

to install it several times, and it's consistently slow at the `Receiving Objects` stage - it takes an ~hour just to get to 19%. The speed varies between several bytes and ~2 kiB/s which is abysmal:

<img width="705" alt="image" src="https://user-images.githubusercontent.com/62620821/191831093-458d2423-7f6c-4101-854d-3d7f8b4ff00f.png">

So, I have hard time installing it. I'm using M1 Macbook. I would appreciate any help.

**Comment by KernelBypass** [comment, marked as answer]
I wonder if it could be connected with GitHub's degraded perf issues today since the shell script is pulling stuff from GH cdn:

<img width="1685" alt="image" src="https://user-images.githubusercontent.com/62620821/191832920-fa029a11-c5b5-40e1-8895-e62ffe350b49.png">

**Reply by SMillerDev** [reply]
That could be, brew is all git repos underneath.

**Reply by KernelBypass** [reply]
Yup, the problem resolved itself.
The issue was probably the GitHub.

**Comment by Citcatjoe** [comment]
Hello.

I just got my macbook pro M1 Max on macOS Monterey at work... and i am trying to install home brew. I am having this this issue during the installation. 

So, installation is going on and suddently stops at "Receiving objects" : 

`==> Downloading and installing Homebrew...
remote: Enumerating objects: 219384, done.
remote: Counting objects: 100% (99/99), done.
remote: Compressing objects: 100% (84/84), done.
Receiving objects:   0% (1/219384)`

And then.. nothing... i mean... i waited for minutes without seeing any progress...
Reading your post, should i just wait and try later or is there something i could try ? Thank you.

**Reply by zhazi0101** [reply]
@Citcatjoe , did you get this issue fixed? I am having a similar issue except it got stuck at 4% of "Receiving objects".

**Reply by HelenYue** [reply]
I got the same issue...tried several times, but still stops at "Receiving objects"...any advise?

**Reply by emmahboss** [reply]
Getting the same error as I speak, Any solution for this yet? Anyone solving this for me has to be my guardian angel.

**Reply by Ethan0429** [reply]
## If you're looking for a solution...

there isn't one. As the devs have said, homebrew is built entirely on git. All of the stuff you download is from github. So if github cdn is performing poorly, so will download speeds. Just a bad time.

**Reply by zhazi0101** [reply]
@HelenYue , @emmahboss , in my case, the firewall is the reason blocking the downloading process. Once I disabled the firewall (the Mcafee installed by company), it downloaded fine. Check it on your side.

**Reply by HelenYue** [reply]
I agree. Maybe network. In my case, I restarted my laptop and everything seems good since then.

**Reply by elalexnc** [reply]
I deactivated my VPN and THAT fixed the stall!!!

**Comment by Trinxic** [comment]
Most likely a firewall issue. Once I connected to my phone's hotspot, the issue went away
